src/main/java/com/synopsys/integration/alert/web/config/FieldModelProcessor.java

```diff
@@ -56,72 +56,94 @@
     }
 
     //TODO: revisit the API of this class because we use a mix of objects. FieldModel and ConfigurationModel here.  Is that correct.
     public FieldModel performAfterReadAction(final FieldModel fieldModel) {
-        return fireEvent(fieldModel, ConfigurationEventType.CONFIG_GET_AFTER);
+        return retrieveApiAction(fieldModel)
+                   .map(apiAction -> apiAction.afterGetAction(fieldModel))
+                   .orElse(fieldModel);
     }
 
     public FieldModel performBeforeDeleteAction(final ConfigurationModel configurationModel) throws AlertDatabaseConstraintException {
         final FieldModel fieldModel = convertToFieldModel(configurationModel);
-        return fireEvent(fieldModel, ConfigurationEventType.CONFIG_DELETE_BEFORE);
+        return retrieveApiAction(fieldModel)
+                   .map(apiAction -> apiAction.beforeDeleteAction(fieldModel))
+                   .orElse(fieldModel);
     }
 
     public void performAfterDeleteAction(final String descriptorName, final String context) {
-        fireEvent(descriptorName, context, ConfigurationEventType.CONFIG_DELETE_AFTER);
+        retrieveApiAction(descriptorName, context).ifPresent(apiAction -> apiAction.afterDeleteAction(descriptorName, context));
     }
 
     public FieldModel performBeforeSaveAction(final FieldModel fieldModel) {
-        return fireEvent(fieldModel, ConfigurationEventType.CONFIG_SAVE_BEFORE);
+        return retrieveApiAction(fieldModel)
+                   .map(apiAction -> apiAction.beforeSaveAction(fieldModel))
+                   .orElse(fieldModel);
     }
 
     public FieldModel performAfterSaveAction(final FieldModel fieldModel) {
-        return fireEvent(fieldModel, ConfigurationEventType.CONFIG_SAVE_AFTER);
+        return retrieveApiAction(fieldModel)
+                   .map(apiAction -> apiAction.afterSaveAction(fieldModel))
+                   .orElse(fieldModel);
     }
 
     public FieldModel performBeforeUpdateAction(final FieldModel fieldModel) {
-        return fireEvent(fieldModel, ConfigurationEventType.CONFIG_UPDATE_BEFORE);
+        return retrieveApiAction(fieldModel)
+                   .map(apiAction -> apiAction.beforeUpdateAction(fieldModel))
+                   .orElse(fieldModel);
     }
 
     public FieldModel performAfterUpdateAction(final FieldModel fieldModel) {
-        return fireEvent(fieldModel, ConfigurationEventType.CONFIG_UPDATE_AFTER);
+        return retrieveApiAction(fieldModel)
+                   .map(apiAction -> apiAction.afterUpdateAction(fieldModel))
+                   .orElse(fieldModel);
     }
 
-    private FieldModel fireEvent(final FieldModel fieldModel, final ConfigurationEventType configurationEventType) {
-        configurationEventPublisher.publishConfigurationEvent(fieldModel, configurationEventType);
-        return fieldModel;
+    public Optional<ApiAction> retrieveApiAction(final FieldModel fieldModel) {
```

Comment:
Since these retrieve methods seem to be used internally in the FieldModelProcessor and no where else make the methods private.  Or if you want to leave them available to sub-classes of FieldModelProcessor make them protected.